src/analyses/expsplit.ml

```diff
@@ -46,10 +46,13 @@ struct
   let return ctx (exp:exp option) (f:fundec) =
     emit_splits_ctx ctx
 
-  let combine ctx (lval:lval option) fexp (f:fundec) (args:exp list) fc au (f_ask: Queries.ask) =
+  let combine_env ctx lval fexp f args fc au f_ask =
     let d = D.join ctx.local au in
     emit_splits ctx d
 
+  let combine_assign ctx (lval:lval option) fexp (f:fundec) (args:exp list) fc au (f_ask: Queries.ask) =
+    emit_splits_ctx ctx
```

Comment:
Is it necessary to do `emit_splits` twice here?

Reply:
I wondered it too and thought there could be some corner case where it matters, but didn't spend time trying to construct it.
The idea is that after `combine_env` there could be splits over global variables, which need to be updated/preserved, and after `combine_assign` there could be a split over the assigned variable, which needs to be updated/preserved. Maybe when the assignment is to a global variable which was also modified inside the function (but its new value comes from the return value)?

Reply:
I think as long as we are convinced that it might make sense to have to do it twice, it doesn't hurt to do it twice. Probably not worth investing too much time in an example where it matters.